.pre-commit-config.yaml

```diff
@@ -84,13 +84,6 @@ repos:
         entry: .hooks/go-vet.sh
         files: '\.go$'
 
-      - id: go-critic
```

Comment:
go-critic appears to be buggy and report erroneous errors. It flags this code: 

```
// However, certain step types (especially SubTTPs) need to run cleanup even if they fail
func (s *Step) ShouldCleanupOnFailure() bool {
	if _, ok := s.action.(*SubTTPStep); ok {
		return true
	}
	return false
}
```

With this message: 

```
go-critic................................................................Failed
- hook id: go-critic
- exit code: 1

./pkg/blocks/step.go:75:14: sloppyTypeAssert: type assertion from/to types are identical
./pkg/blocks/step.go:87:14: sloppyTypeAssert: type assertion from/to types are identical
```

This is not true - s.action is of type `Action`, which is an interface and is not identical to `*SubTTPStep`, which is a pointer to a struct. 

Furthermore, when the code in question is rewritten to use a switch, go-critic actually just can't figure out the type: 

```
./pkg/blocks/step.go:90:2: caseOrder: type is not defined *SubTTPStep
``` 

All of this code works and passes its appropriate unit tests, which explicitly target the case in question. 

We already have `go lint`, and `go-critic` has too many of these issues to be worth keeping.